src/FRP/Event/Keyboard.js

```diff
@@ -1,20 +1,20 @@
 "use strict";
 
-var currentKeys = [];
+var currentKeys = {};
 addEventListener("keydown", function(e) {
-  currentKeys.push(e.keyCode);
+  currentKeys[e.keyCode] = true;
 });
 addEventListener("keyup", function(e) {
-  var index = currentKeys.indexOf(e.keyCode);
-  if (index >= 0) {
-    currentKeys.splice(index, 1);
-  }
+  currentKeys[e.keyCode] = false;
 });
 
 exports.withKeys = function (e) {
   return function(sub) {
     return e(function(a) {
-      sub({ keys: currentKeys, value: a });
+      var currentKeysArray = Object.keys(currentKeys)
```

Comment:
Could you use `Array.prototype.keys` here and avoid the `parseInt`?

Reply:
I'm not sure what you mean. That doesn't seem to give the correct result.

From [Mozilla](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/keys)

> The keys() method returns a new Array Iterator object that contains the keys for each index in the array.

Given this example:

~~~ javascript
>>> const a = {12: 'a', 17: 'b', 38: 'c', 2: 'd'}
>>> Object.keys(a)
["2", "12", "17", "38"]
~~~

I get the following:

~~~ javascript
>>> Array.from(Object.keys(a).keys())
[0, 1, 2, 3]

>>> [...Object.keys(a).keys()]
[0, 1, 2, 3]
~~~

But I want to get:

~~~ javascript
[2, 12, 17, 38]
~~~

Which the current implementation provides.

I don't care for the `parseInt` either, but I can't think of a simpler way to do the same thing.

Reply:
Sorry, yes, that was a bad suggestion. My point though is that the keys are integers, so you can just look from `0` to `currentKeysArray.length - 1` and see which ones contain `true` (being careful to avoid `undefined`).

Reply:
@paf31 I'm not sure what you mean. Could you provide a code example?

Reply:
Something like

```js
var currentKeys = [];
addEventListener("keydown", function(e) {
  currentKeys[e.keyCode] = true;
});
addEventListener("keyup", function(e) {
  currentKeys[e.keyCode] = false;
});

exports.withKeys = function (e) {
  return function(sub) {
    return e(function(a) {
      var keys = [];
      for (var i = 0; i < currentKeys.length; i++) {
        if (currentKeys[i] === true) {
          keys.push(i);
        }
      }
      sub({ keys: keys, value: a });
    });
  };
};
```

Reply:
I see. This assumes that `currentKeys` is an Array, but It is a plain Object in this PR. I think this change will work fine, but am curious if the gaps will cause any issues at some point.

Exactly what version of javascript do we have available for FFI modules?

Reply:
I'd prefer ES3 if possible, but ES5 is fine if necessary.